includes/apple-exporter/components/class-podcast.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Publish to Apple News Includes: Apple_Exporter\Components\Podcast class
+ *
+ * Contains a class which is used to transform Podcast embeds into Apple News format.
+ * Documentation about apple podcast players can be found here: https://podcasters.apple.com/support/889-apple-podcasts-embed-player.
+ *
+ * @package Apple_News
+ * @subpackage Apple_Exporter
+ * @since 0.2.0
+ */
+
+namespace Apple_Exporter\Components;
+
+/**
+ * A class to transform a Podcast embed into an Apple News Podcast component.
+ *
+ * @since 0.2.0
+ */
+class Podcast extends Component {
+
+	/**
+	 * Look for node matches for this component.
+	 *
+	 * @param \DOMElement $node The node to examine for matches.
+	 * @access public
+	 * @return \DOMElement|null The node on success, or null on no match.
+	 */
+	public static function node_matches( $node ) {
+		// Handle node if Podcast url is present.
+		if ( self::get_podcast_url( $node ) ) {
+			return $node;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Register all specs for the component.
+	 *
+	 * @access public
+	 */
+	public function register_specs() {
+		$this->register_spec(
+			'json',
+			__( 'JSON', 'apple-news' ),
+			[
+				'role' => 'podcast',
+				'URL'  => '#url#',
+			]
+		);
+	}
+
+	/**
+	 * Build the component.
+	 *
+	 * @param string $html The HTML to parse into text for processing.
+	 * @access protected
+	 */
+	protected function build( $html ) {
+		// Try to get podcast URL.
+		$url = self::get_podcast_url( $html );
+
+		// Ensure we got a URL.
+		if ( empty( $url ) ) {
+			return;
+		}
+
+		$this->register_json(
+			'json',
+			[
+				'#url#' => esc_url_raw( $url ),
+			]
+		);
+	}
+
+	/**
+	 * A method to get an Podcast URL from provided node.
+	 *
+	 * @param string $node The node to parse for the Podcast URL.
+	 *
+	 * @access private
+	 * @return string|false The Podcast URL on success, or false on failure.
+	 */
+	private static function get_podcast_url( $node ) {
+		if ( 'string' === gettype( $node ) ) {
+			$dom = new \DomDocument();
+			$dom->loadHTML( $node );
+			$node = $dom;
+		}
+
+		$iframe = self::get_iframe_from_node( $node );
+
+		if ( ! $iframe ) {
+			return;
+		}
+
+		// Pattern match src attribute for apple podcast url.
+		$url = $iframe->getAttribute( 'src' );
+
+		if ( empty( $url ) || ! str_contains( $url, 'podcasts.apple.com' ) ) {
+			return;
+		}
+
+		// Remove iframe specific `embed.` prefix on podcast url.
+		$url = preg_replace( '/(embed.)(podcasts\.apple\.com)/', '$2', $url );
```

Comment:
Can be simplified to a `str_replace` which runs faster than `preg_replace` and is easier to read:
```suggestion
		$url = str_replace( 'embed.podcasts.apple.com', 'podcasts.apple.com', $url );
```